tracy-client/src/lib.rs

```diff
@@ -51,18 +51,18 @@ pub mod internal {
     use std::ffi::CString;
     pub use std::ptr::null;
 
-    #[inline(always)]
+    #[must_use]
```

Comment:
Unless there is a good reason not to (e.g. measured performance is worse,) lets keep the  `#[inline(always)]`.

Reply:
Did you benchmark their use? The default should be to **not** use `inline(always)`: https://rust-lang.github.io/rust-clippy/master/index.html#/inline_always. I've also had folks from the stdlib team tell me that it slows down compile times and bloats codegen if llvm thinks using a function would actually be better. Basically the rule is that you need to have measured a perf improvement before using `inline(always)`.

Reply:
I do recall looking at the assembly way back for some of these. It might very well be the case that after the many LLVM upgrades the situation has changed, or it might be that it hasn’t, I don’t really have the time to look into this right now.

---

That said, I can motivate why these *should* be `#[inline(always)]` even outside of the “measure-and-prove” line of reasoning.

Particularly worth noting is that many of these `internal` functions are just basic wrappers over otherwise private constructors to be used in this crate’s macros. Constructing an ADT is pretty much *always* simpler than making a function call to do the same, no matter how many uses of this operation there are.

This is especially notable in the context of the compile-time blowup reasoning that clippy presents as a counter-argument. It makes a lot of sense when we’re talking about functions with a non-trivial amount of logic and such. It does not hold at all for wrapper functions like these (and clippy itself presents some exceptions such as functions that have nothing in their body or simply panic.)

If it wasn’t for the limitations in the Rust’s visibility system in combination with macros, these functions wouldn’t exist at all. In that sense `#[inline(always)]` serves as a pretty good mechanism to also express that sort of intent.

Finally `inline(always)` has a benefit of achieving these intents even in debug builds.

Reply:
I think that makes sense? It still leaves me uneasy lol. Anyway, I brought back the `inline(always)` attributes for the `mod internal` methods.

Did you want all the inlines brought back? I'll disagree a bit more strongly about the others.